driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/pageobjects/internal/ResponsiveClassFinder.java

```diff
@@ -57,7 +62,12 @@ final public class ClassFinder {
     private static final String PATTERN_HI = SCHEMA_DIV + RESOLUTION_REGEX + SCHEMA_DIV + KEYWORD_MAX;
     private static final String PATTERN_RES = "TODO"; // TODO
 
-    private ClassFinder() {
+    /**
+     * This call takes some time. It has an impact to the duration of the first page check (takes ca 2-3 seconds longer).
+     */
+    private static final Reflections reflections = new Reflections(new ConfigurationBuilder().setUrls(ClasspathHelper.forJavaClassPath()));
```

Comment:
Whoa! 2-3 seconds!? I am somewhat surprised by that.

Maybe we should investigate whether we can filter harder in the ConfigurationBuilder, e.g. using `#setScanners` to restrict the kind of resource we are looking for, or `#filterInputsBy` to fail-fast on certain kinds of resource (maybe we can say that some base classes _disqualify_ a class as a candidate?).

Reply:
We do not know the package of the test project, so we cannot restrict it to defined packages. 

The filtering with `new ConfigurationBuilder()....filterInputsBy(new FilterBuilder().??)` seems to work only at packages, not on classes/subclasses.

Any other filtering by subtypes are only possible in a second step with `reflections.getSubTypesOf()`.

Reply:
> We do not know the package of the test project, so we cannot restrict it to defined packages

Is this something that we could set via a test.property or do we prefer the extra wait time on test execution over adding something to the required setup?

Reply:
Exactly this property will be removed by this PR to provide more comfort. 

I think we can handle this extra time because with this change it is only occured once.

Reply:
... We likely can and should optimise the call `ClasspathHelper.forJavaClassPath()`.

This is what it produces when I run a minimal test locally:
```
0 = {URL@2307} "file:/C:/Program%20Files/JetBrains/IntelliJ%20IDEA%202021.3.2/lib/idea_rt.jar"
1 = {URL@2308} "file:/C:/Program%20Files/JetBrains/IntelliJ%20IDEA%202021.3.2/plugins/testng/lib/testng-rt.jar"
2 = {URL@2309} "file:/C:/Users/smby/Projekte/testerra/driver-ui/build/classes/java/test/"
3 = {URL@2310} "file:/C:/Users/smby/Projekte/testerra/driver-ui/build/classes/java/main/"
4 = {URL@2311} "file:/C:/Users/smby/Projekte/testerra/driver-ui/build/resources/main/"
5 = {URL@2312} "file:/C:/Users/smby/Projekte/testerra/core/build/classes/java/main/"
6 = {URL@2313} "file:/C:/Users/smby/Projekte/testerra/core/build/resources/main/"
7 = {URL@2314} "file:/C:/Users/smby/.m2/repository/org/sikuli/sikuli-api/1.2.0/sikuli-api-1.2.0.jar"
8 = {URL@2315} "file:/C:/Users/smby/.m2/repository/org/seleniumhq/selenium/selenium-java/3.141.59/selenium-java-3.141.59.jar"
9 = {URL@2316} "file:/C:/Users/smby/.m2/repository/org/bytedeco/javacpp-presets/opencv/2.4.9-0.9/opencv-2.4.9-0.9.jar"
10 = {URL@2317} "file:/C:/Users/smby/.m2/repository/org/bytedeco/javacpp-presets/opencv/2.4.9-0.9/opencv-2.4.9-0.9-linux-x86.jar"
11 = {URL@2318} "file:/C:/Users/smby/.m2/repository/org/bytedeco/javacpp-presets/opencv/2.4.9-0.9/opencv-2.4.9-0.9-linux-x86_64.jar"
12 = {URL@2319} "file:/C:/Users/smby/.m2/repository/org/bytedeco/javacpp-presets/opencv/2.4.9-0.9/opencv-2.4.9-0.9-windows-x86.jar"
13 = {URL@2320} "file:/C:/Users/smby/.m2/repository/org/bytedeco/javacpp-presets/opencv/2.4.9-0.9/opencv-2.4.9-0.9-windows-x86_64.jar"
14 = {URL@2321} "file:/C:/Users/smby/.m2/repository/nu/pattern/opencv/2.4.9-7/opencv-2.4.9-7.jar"
15 = {URL@2322} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.google.code.gson/gson/2.8.6/9180733b7df8542621dc12e21e87557e8c99b8cb/gson-2.8.6.jar"
16 = {URL@2323} "file:/C:/Users/smby/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar"
17 = {URL@2324} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.reflections/reflections/0.9.12/1c9d44c563eebe9b8a3afebd29ed5c4646db800c/reflections-0.9.12.jar"
18 = {URL@2325} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.github.ua-parser/uap-java/1.4.3/a457ec671f02064ca6119673699a52eaf576d81f/uap-java-1.4.3.jar"
19 = {URL@2326} "file:/C:/Users/smby/Projekte/testerra/logging/build/classes/java/main/"
20 = {URL@2327} "file:/C:/Users/smby/Projekte/testerra/logging/build/resources/main/"
21 = {URL@2328} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.google.inject.extensions/guice-assistedinject/4.2.2/c33fb10080d58446f752b4fcfff8a5fabb80a449/guice-assistedinject-4.2.2.jar"
22 = {URL@2329} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.google.inject/guice/4.2.2/6dacbe18e5eaa7f6c9c36db33b42e7985e94ce77/guice-4.2.2.jar"
23 = {URL@2330} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.google.inject/guice/4.2.2/fa13659f9128f4c011c8e1d06f137083b4876377/guice-4.2.2-no_aop.jar"
24 = {URL@2331} "file:/C:/Users/smby/.m2/repository/commons-io/commons-io/2.6/commons-io-2.6.jar"
25 = {URL@2332} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.apache.commons/commons-lang3/3.9/122c7cee69b53ed4a7681c03d4ee4c0e2765da5/commons-lang3-3.9.jar"
26 = {URL@2333} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.google.guava/guava/28.1-jre/b0e91dcb6a44ffb6221b5027e12a5cb34b841145/guava-28.1-jre.jar"
27 = {URL@2334} "file:/C:/Users/smby/.m2/repository/org/jsoup/jsoup/1.7.2/jsoup-1.7.2.jar"
28 = {URL@2335} "file:/C:/Users/smby/.m2/repository/org/jdom/jdom2/2.0.6/jdom2-2.0.6.jar"
29 = {URL@2336} "file:/C:/Users/smby/.m2/repository/javax/xml/bind/jaxb-api/2.3.0/jaxb-api-2.3.0.jar"
30 = {URL@2337} "file:/C:/Users/smby/.m2/repository/com/sun/xml/bind/jaxb-core/2.3.0/jaxb-core-2.3.0.jar"
31 = {URL@2338} "file:/C:/Users/smby/.m2/repository/com/sun/xml/bind/jaxb-impl/2.3.0/jaxb-impl-2.3.0.jar"
32 = {URL@2339} "file:/C:/Users/smby/.m2/repository/javax/activation/activation/1.1/activation-1.1.jar"
33 = {URL@2340} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/xerces/xercesImpl/2.8.0/cfd3ebe2f8034e660344f9108c3e2daf78c29cc3/xercesImpl-2.8.0.jar"
34 = {URL@2341} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.testng/testng/7.4.0/b6becc0a058e272473ac3ba983966ff507ca0300/testng-7.4.0.jar"
35 = {URL@2342} "file:/C:/Users/smby/.m2/repository/org/sikuli/sikuli-core/1.2.2/sikuli-core-1.2.2.jar"
36 = {URL@2343} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.slf4j/slf4j-api/1.8.0-beta4/83b0359d847ee053d745be7ec0d8e9e8a44304b4/slf4j-api-1.8.0-beta4.jar"
37 = {URL@2344} "file:/C:/Users/smby/.m2/repository/org/seleniumhq/selenium/selenium-chrome-driver/3.141.59/selenium-chrome-driver-3.141.59.jar"
38 = {URL@2345} "file:/C:/Users/smby/.m2/repository/org/seleniumhq/selenium/selenium-edge-driver/3.141.59/selenium-edge-driver-3.141.59.jar"
39 = {URL@2346} "file:/C:/Users/smby/.m2/repository/org/seleniumhq/selenium/selenium-firefox-driver/3.141.59/selenium-firefox-driver-3.141.59.jar"
40 = {URL@2347} "file:/C:/Users/smby/.m2/repository/org/seleniumhq/selenium/selenium-ie-driver/3.141.59/selenium-ie-driver-3.141.59.jar"
41 = {URL@2348} "file:/C:/Users/smby/.m2/repository/org/seleniumhq/selenium/selenium-opera-driver/3.141.59/selenium-opera-driver-3.141.59.jar"
42 = {URL@2349} "file:/C:/Users/smby/.m2/repository/org/seleniumhq/selenium/selenium-safari-driver/3.141.59/selenium-safari-driver-3.141.59.jar"
43 = {URL@2350} "file:/C:/Users/smby/.m2/repository/org/seleniumhq/selenium/selenium-support/3.141.59/selenium-support-3.141.59.jar"
44 = {URL@2351} "file:/C:/Users/smby/.m2/repository/org/seleniumhq/selenium/selenium-remote-driver/3.141.59/selenium-remote-driver-3.141.59.jar"
45 = {URL@2352} "file:/C:/Users/smby/.m2/repository/org/seleniumhq/selenium/selenium-api/3.141.59/selenium-api-3.141.59.jar"
46 = {URL@2353} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/net.bytebuddy/byte-buddy/1.8.15/cb36fe3c70ead5fcd016856a7efff908402d86b8/byte-buddy-1.8.15.jar"
47 = {URL@2354} "file:/C:/Users/smby/.m2/repository/org/apache/commons/commons-exec/1.3/commons-exec-1.3.jar"
48 = {URL@2355} "file:/C:/Users/smby/.m2/repository/com/squareup/okhttp3/okhttp/3.11.0/okhttp-3.11.0.jar"
49 = {URL@2356} "file:/C:/Users/smby/.m2/repository/com/squareup/okio/okio/1.14.0/okio-1.14.0.jar"
50 = {URL@2357} "file:/C:/Users/smby/.m2/repository/org/bytedeco/javacpp/0.9/javacpp-0.9.jar"
51 = {URL@2358} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.javassist/javassist/3.26.0-GA/bb2890849968d8d8311ffba8c37b0ce16ce284dc/javassist-3.26.0-GA.jar"
52 = {URL@2359} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.yaml/snakeyaml/1.20/11e7e64e621e5e43c7481bf01072a7b1597d4f03/snakeyaml-1.20.jar"
53 = {URL@2360} "file:/C:/Users/smby/.m2/repository/org/apache/commons/commons-collections4/4.1/commons-collections4-4.1.jar"
54 = {URL@2361} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-slf4j18-impl/2.17.2/79462d0451b93106225395b03362b1965984533b/log4j-slf4j18-impl-2.17.2.jar"
55 = {URL@2362} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-core/2.17.2/fa43ba4467f5300b16d1e0742934149bfc5ac564/log4j-core-2.17.2.jar"
56 = {URL@2363} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-api/2.17.2/f42d6afa111b4dec5d2aea0fe2197240749a4ea6/log4j-api-2.17.2.jar"
57 = {URL@2364} "file:/C:/Users/smby/.m2/repository/javax/inject/javax.inject/1/javax.inject-1.jar"
58 = {URL@2365} "file:/C:/Users/smby/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar"
59 = {URL@2366} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.google.guava/failureaccess/1.0.1/1dcf1de382a0bf95a3d8b0849546c88bac1292c9/failureaccess-1.0.1.jar"
60 = {URL@2367} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.google.guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/b421526c5f297295adef1c886e5246c39d4ac629/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar"
61 = {URL@2368} "file:/C:/Users/smby/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar"
62 = {URL@2369} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.checkerframework/checker-qual/2.8.1/eb2e8ab75598548cc8acf9a1ca227e480e01881e/checker-qual-2.8.1.jar"
63 = {URL@2370} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.google.errorprone/error_prone_annotations/2.3.2/d1a0c5032570e0f64be6b4d9c90cdeb103129029/error_prone_annotations-2.3.2.jar"
64 = {URL@2371} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.google.j2objc/j2objc-annotations/1.3/ba035118bc8bac37d7eff77700720999acd9986d/j2objc-annotations-1.3.jar"
65 = {URL@2372} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.codehaus.mojo/animal-sniffer-annotations/1.18/f7aa683ea79dc6681ee9fb95756c999acbb62f5d/animal-sniffer-annotations-1.18.jar"
66 = {URL@2373} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/xml-apis/xml-apis/1.3.03/3845d5aabd62dc1954f2c0e84a799068c917ad2b/xml-apis-1.3.03.jar"
67 = {URL@2374} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.beust/jcommander/1.78/a3927de9bd6f351429bcf763712c9890629d8f51/jcommander-1.78.jar"
68 = {URL@2375} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.webjars/jquery/3.5.1/2392938e374f561c27c53872bdc9b6b351b6ba34/jquery-3.5.1.jar"
69 = {URL@2376} "file:/C:/Users/smby/.m2/repository/org/piccolo2d/piccolo2d-extras/1.3.1/piccolo2d-extras-1.3.1.jar"
70 = {URL@2377} "file:/C:/Users/smby/.m2/repository/org/piccolo2d/piccolo2d-core/1.3.1/piccolo2d-core-1.3.1.jar"
71 = {URL@2378} "file:/C:/Users/smby/.m2/repository/com/opencsv/opencsv/3.9/opencsv-3.9.jar"
72 = {URL@2379} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.apache.pdfbox/pdfbox/1.8.16/3ee5ba356834cb1899c84b40ce1aabc24ae6db25/pdfbox-1.8.16.jar"
73 = {URL@2380} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.apache.poi/poi-ooxml/3.17/7d8c44407178b73246462842bf1e206e99c8e0a/poi-ooxml-3.17.jar"
74 = {URL@2381} "file:/C:/Users/smby/.m2/repository/net/lingala/zip4j/zip4j/1.3.2/zip4j-1.3.2.jar"
75 = {URL@2382} "file:/C:/Users/smby/.m2/repository/de/idyl/winzipaes/1.0.1/winzipaes-1.0.1.jar"
76 = {URL@2383} "file:/C:/Users/smby/.m2/repository/commons-beanutils/commons-beanutils/1.9.3/commons-beanutils-1.9.3.jar"
77 = {URL@2384} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.apache.pdfbox/fontbox/1.8.16/a4aece226b0412bd459ed3290b91d98a3f1c6d2e/fontbox-1.8.16.jar"
78 = {URL@2385} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.apache.pdfbox/jempbox/1.8.16/1f41de81768ef84ca2d8cda4cb79e9272c8ee966/jempbox-1.8.16.jar"
79 = {URL@2386} "file:/C:/Users/smby/.m2/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar"
80 = {URL@2387} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.apache.poi/poi/3.17/ae92292a2043888b40d418da97dc0b669fde326/poi-3.17.jar"
81 = {URL@2388} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.apache.poi/poi-ooxml-schemas/3.17/890114bfa82f5b6380ea0e9b0bf49b0af797b414/poi-ooxml-schemas-3.17.jar"
82 = {URL@2389} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/com.github.virtuald/curvesapi/1.04/3386abf821719bc89c7685f9eaafaf4a842f0199/curvesapi-1.04.jar"
83 = {URL@2390} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.bouncycastle/bcprov-jdk16/1.45/612d4afcd33f05a5ffdd9cf835e3bd4e3b44018/bcprov-jdk16-1.45.jar"
84 = {URL@2391} "file:/C:/Users/smby/.m2/repository/commons-codec/commons-codec/1.10/commons-codec-1.10.jar"
85 = {URL@2392} "file:/C:/Users/smby/.gradle/caches/modules-2/files-2.1/org.apache.xmlbeans/xmlbeans/2.6.0/29e80d2dd51f9dcdef8f9ffaee0d4dc1c9bbfc87/xmlbeans-2.6.0.jar"
86 = {URL@2393} "file:/C:/Users/smby/.m2/repository/stax/stax-api/1.0.1/stax-api-1.0.1.jar"
```

Surely we could filter out at least `.m2` and `.gradle` - but that might break silently in the future, so could we perhaps offer a _whitelist_ ?

Reply:
The default ConfigurationBuilder will also have two Scanners: `TypeAnnotationScanner` and `SubTypesScanner`.
Given that we only ever call `getSubTypesOf` , we can probably explicitly drop the first one.

Reply:
> We do not know the package of the test project, so we cannot restrict it to defined packages.
> 
> The filtering with `new ConfigurationBuilder()....filterInputsBy(new FilterBuilder().??)` seems to work only at packages, not on classes/subclasses.
> 
> Any other filtering by subtypes are only possible in a second step with `reflections.getSubTypesOf()`.

No, it works on each and every found resource. Case in point, I ran this:
```java 
int[] countXsb = {0}, countMF = {0}, countXML = {0}, countProperties = {0};
        configuration.filterInputsBy(name -> {
            if (name.endsWith(".xml"))
                ++countXML[0];
            else if (name.endsWith(".MF"))
                ++countMF[0];
            else if (name.endsWith(".xsb"))
                ++countXsb[0];
            else if (name.endsWith(".properties"))
                ++countProperties[0];
            System.err.println(name);
            return true;
        });
```
And it counted the following:

- 3697 files ending with ".xsb"
- 78 files ending with ".MF"
- 63 files ending with ".xml"
- 229 files ending with ".properties"

We should filter for `.filterInputsBy(name -> name.endsWith(".class"))`.

Reply:
`filterInputs` seems to be executed _after_ scanning all ressources.

Using the option `useParallelExecutor` improves the performance:

````java
new ConfigurationBuilder()
                        .setUrls(ClasspathHelper.forJavaClassPath())
                        .useParallelExecutor()
                        .setScanners(new SubTypesScanner())
                        .filterInputsBy(name -> name.endsWith(".class"))
````